File: 卡码/代码随想录/Go/贪心/02-中等/LeetCode_0122.go
//122. 买卖股票的最佳时机 II
package main

func maxProfit(prices []int) int {
	res:=0
	for i:=1;i<len(prices);i++ {
		tmp:=prices[i]-prices[i-1]
		if tmp>0 {
			res+=tmp
		}
	}
	return res
}

File: 卡码/代码随想录/Python/贪心/02-中等/LeetCode_0122.py
#122. 买卖股票的最佳时机 II
from typing import *

class Solution:
  def maxProfit(self, prices: List[int]) -> int:
    res:int=0
    for i in range(1,len(prices)):
      res+=max(prices[i]-prices[i-1],0)
    return res